nixos/modules/services/development/lorri.nix

```
@@ -0,0 +1,56 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.lorri;
+  socketUnit = "lorri";
+  socketPath = "lorri/daemon.socket";
+in {
+  options = {
+    services.lorri = {
+      enable = lib.mkOption {
+        default = false;
+        type = lib.types.bool;
+        description = ''
+          This option enables a systemd socket unit that listens on the
+          well-known address used by lorri, a nix-shell replacement for project
+          development. When a lorri client connects to the socket, systemd
+          starts the lorri daemon as a service.
+        '';
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.user.sockets.${socketUnit} = {
+      description = "Socket for Lorri Daemon";
+      wantedBy = [ "sockets.target" ];
+      unitConfig = {
+        ConditionUser = "!@system";
+      };
+      socketConfig = {
+        ListenStream = "%t/${socketPath}";
```

```suggestion
        ListenStream = "%t/lorri/daemon.socket";
```
Again, I'd inline `${socketPath}`, this time for readability

---

See https://github.com/NixOS/nixpkgs/pull/72889#discussion_r345119285 - I think the variable makes the intention clearer.

---

See my argument above on why I think we shouldn't use the `${socketPath}` variable, but just hardcode things in here.

---

Changed